keras/applications/inception_resnet_v1.py

```diff
@@ -0,0 +1,379 @@
+                                   block_idx=block_idx)
+
+    # Mixed 7a (Reduction-B block):
+    branch_0 = conv2d_bn(x, 256, 1)
+    branch_0 = conv2d_bn(branch_0,
+                         384,
+                         3,
+                         strides=2,
+                         padding='valid')
+    branch_1 = conv2d_bn(x, 256, 1)
+    branch_1 = conv2d_bn(branch_1,
+                         256,
+                         3,
+                         strides=2,
+                         padding='valid')
+    branch_2 = conv2d_bn(x, 256, 1)
+    branch_2 = conv2d_bn(branch_2, 256, 3)
+    branch_2 = conv2d_bn(branch_2,
+                         256,
+                         3,
+                         strides=2,
+                         padding='valid')
+    branch_pool = MaxPooling2D(3,
+                               strides=2,
+                               padding='valid')(x)
+    branches = [branch_0, branch_1, branch_2, branch_pool]
+    x = Concatenate(axis=channel_axis, name='Mixed_7a')(branches)
+
+    # 5x Block8 (Inception-ResNet-C block):
+    for block_idx in range(1, 6):
+        x = inception_resnet_block(x,
+                                   scale=0.2,
+                                   block_type='block8',
+                                   block_idx=block_idx)
+
+    x = inception_resnet_block(x,
+                               scale=1.,
+                               activation=None,
+                               block_type='block8',
+                               block_idx=6)
+
+    if include_top:
+        # Classification block
+        x = GlobalAveragePooling2D(name='avg_pool')(x)
+        x = Dense(classes, activation='softmax', name='predictions')(x)
+    else:
+        if pooling == 'avg':
+            x = GlobalAveragePooling2D()(x)
+        elif pooling == 'max':
+            x = GlobalMaxPooling2D()(x)
+
+    # Ensure that the model takes into account
+    # any potential predecessors of `input_tensor`
+    if input_tensor is not None:
+        inputs = get_source_inputs(input_tensor)
+    else:
+        inputs = img_input
+
+    # Create model
+    model = Model(inputs, x, name='inception_resnet_v1')
+
+    # Load weights
+    if weights == 'facenet':
+        if K.image_data_format() == 'channels_first':
+            if K.backend() == 'tensorflow':
+                warnings.warn('You are using the TensorFlow backend, yet you '
+                              'are using the Theano '
+                              'image data format convention '
+                              '(`image_data_format="channels_first"`). '
+                              'For best performance, set '
+                              '`image_data_format="channels_last"` in '
+                              'your Keras config '
+                              'at ~/.keras/keras.json.')
+        if include_top:
+            raise ValueError('currently does not support top')
```

Comment:
Why? Also the error message is not correct (the model can be created with `include_top=True, weights=None`) and sounds a bit weird.